apps/wallet-mobile/src/yoroi-wallets/cardano/assetUtils/assetUtils.ts

```diff
@@ -1,19 +1,24 @@
 import BigNumber from 'bignumber.js'
 
-import {NetworkId, RawUtxo} from '../../types/other'
+import {Address} from '../../types'
+import {RawUtxo} from '../../types/other'
 import {CardanoMobile} from '../../wallets'
-import {getCardanoNetworkConfigById} from '../networks'
-import {cardanoValueFromRemoteFormat} from '../utils'
+import {COINS_PER_UTXO_BYTE} from '../constants/common'
+import {cardanoValueFromRemoteFormat, normalizeToAddress} from '../utils'
 
-export async function calcLockedDeposit(utxos: RawUtxo[], networkId: NetworkId) {
-  const networkConfig = getCardanoNetworkConfigById(networkId)
-  const coinsPerUtxoWord = await CardanoMobile.BigNum.fromStr(networkConfig.COINS_PER_UTXO_WORD)
+export async function calcLockedDeposit(utxos: RawUtxo[], address: Address, coinsPerUtxoByteStr: string) {
   const utxosWithAssets = utxos.filter((u) => u.assets.length > 0)
 
+  const coinsPerUtxoByte = await CardanoMobile.BigNum.fromStr(coinsPerUtxoByteStr ?? COINS_PER_UTXO_BYTE)
+  const dataCost = await CardanoMobile.DataCost.newCoinsPerByte(coinsPerUtxoByte)
+  const normalizedAddress = await normalizeToAddress(address)
+  if (normalizedAddress === undefined) throw new Error('calcLockedDeposit::Error not a valid address')
+
   const promises = utxosWithAssets.map((u) => {
     return cardanoValueFromRemoteFormat(u)
-      .then((v) => CardanoMobile.minAdaRequired(v, false, coinsPerUtxoWord))
-      .then((v) => v.toStr())
+      .then((v) => CardanoMobile.TransactionOutput.new(normalizedAddress, v))
```

Comment:
Do we 100% that utxo doesn't  have an inlined datum/script ?

Reply:
I think the API doesn't return it now. cc: @pedromtcosta

Reply:
@stackchain  It's not case for casual wallet user, but anyway from blockchain perspective it is possible. If it worked before it will work for now,  but this edge case still will not be covered